test/CAS/mccas.swift

```diff
@@ -0,0 +1,42 @@
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: %target-swift-frontend -frontend -c -primary-file %s -target arm64-apple-darwin23.0.0 -g -cas-backend -cas-backend-mode=verify -cas-path %t/cas -allow-unstable-cache-key-for-testing -cache-compile-job -o %t/test.o
```

Comment:
should the `-mccas` option not dependent on `-cache-compile-job`? I would keep them separate and test both cases (in the future). In your patch, MCCAS is not actually working cache replaying (to support that, swift-frontend needs to correctly set the cached output object as MCCAS object and replay from MCCAS), please drop all `-cache-compile-job` from the tests and change the frontend logic for setup CAS when -cache-compile-job is not used.